cargo/lib/dependabot/cargo/file_fetcher.rb

```diff
@@ -33,6 +33,8 @@ def package_manager_version
             "channel" => channel
           }
         }
+      rescue TomlRB::ParseError
+        raise Dependabot::DependencyFileNotParseable, rust_toolchain.path
```

Comment:
I wish we could pass an optional custom error message in addition to the path, because the rust docs say they still support the legacy format, so we're likely to get customers complaining "`rustc` works with my config file, why doesn't :dependabot: ??"

So this would be a perfect place to say "Rust switched to TOML format since XYZ, so we only support TOML versions"...

Not a big deal though...

Reply:
We can! `Dependabot::DependencyFileNotParseable.new(rust_toolchain.path, <custom_message>)`

Reply:
See https://github.com/dependabot/dependabot-core/pull/7003 (wrong) and https://github.com/dependabot/dependabot-core/pull/7068 (fixed).

Reply:
Actually those PR didn't really fix this, they just give a more sensible message if you don't pass a custom message. But just wanted to say that this should be possible!

Reply:
ah right! I was looking only at https://github.com/dependabot/dependabot-core/blob/009b8f2c34ada42b4a2659b7fd9bd3054049aa78/common/lib/dependabot/errors.rb#L97 but forgot to check the underlying base class.

Reply:
I added a friendlier message. How does that look?